src/bci_build/package/postgres/README.md.j2

```diff
@@ -0,0 +1,114 @@
+
+### Environment variables
+
+The PostgreSQL image uses several environment variables to configure the database initialization.
+
+The only variable required is `POSTGRES_PASSWORD`, all other environment variables are optional.
+
+#### POSTGRES_PASSWORD
+
+This environment variable is required for you to use the PostgreSQL image. It must not be empty or undefined. This environment variable sets the superuser password for PostgreSQL. The default superuser is defined by the `POSTGRES_USER` environment variable.
+
+#### POSTGRES_USER
+
+This optional environment variable is used in conjunction with `POSTGRES_PASSWORD` to set a user and its password. This variable will create the specified user with superuser power and a database with the same name. If it is not specified, then the default user of `postgres` will be used.
+
+#### POSTGRES_DB
+
+This optional environment variable can be used to define a different name for the default database that is created when the image is first started. If it is not specified, then the value of `POSTGRES_USER` will be used.
+
+#### POSTGRES_INITDB_ARGS
+
+This optional environment variable can be used to send arguments to `postgres initdb`. The value is a space-separated string of arguments as `postgres initdb` would expect them. This is useful for adding functionality like data page checksums: `-e POSTGRES_INITDB_ARGS="--data-checksums"`.
+
+#### POSTGRES_INITDB_WALDIR
+
+This optional environment variable can be used to define another location for the Postgres transaction log. By default, the transaction log is stored in a subdirectory of the main Postgres data folder (`PGDATA`). Sometimes it can be desirable to store the transaction log in a different directory which may be backed by storage with different performance or reliability characteristics.
+
+#### POSTGRES_HOST_AUTH_METHOD
+
+This optional variable can be used to control the `auth-method` for host connections for all databases, all users, and all addresses. If unspecified then `scram-sha-256` password authentication is used.
+
+On an uninitialized database, this will populate `pg_hba.conf` via this approximate line:
+
+`echo "host all all all $POSTGRES_HOST_AUTH_METHOD" >> pg_hba.conf`
+
+See the PostgreSQL documentation on [password Authentication](https://www.postgresql.org/docs/14/auth-password.html) and [pg_hba.conf](https://www.postgresql.org/docs/14/auth-pg-hba-conf.html) for more information about possible values and their meanings.
+
+**Note 1:** If you set `POSTGRES_HOST_AUTH_METHOD` to `trust`, then `POSTGRES_PASSWORD` is not required since it allows anyone to connect without a password.
+
+**Note 2:** If you set this to an alternative value, you might need additional `POSTGRES_INITDB_ARGS` for the database to initialize correctly.
+
+#### PGDATA
```

Comment:
we're not supporting changing PGDATA at the moment. it is not being tested in the tests, and I remember we had issues getting it variable working in all circumstances. I would recommend not documenting it for now. 

if you want to keep the text around, just add an `{{ if False }}` or so around it.

Reply:
I was not aware if did not work and I honestly did not test this. I can test, see what happens, possibly create a bug to fix in the future, and omit this bit.

Reply:
There are a few variables from psql server that are not really a container variable, should I explicitly document that these are not supported? This way at least people won't try to use it. 

Namely `PGCLIENTENCODING`, `PGDATA`, `PGDATESTYLE`, `PGPORT`.

One might try to use if we don't document anything and I'm not sure what would be the side effects.

Reply:
we did test it in the past, it did break and we decided to not fix it because it would be different than the configuration that the packages are supported in (in SLE15). 

I'm fine with documenting variables and what their values are as long as we do not suggest to support changing those.